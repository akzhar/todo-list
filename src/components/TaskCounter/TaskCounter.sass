.task-counter
  display: block
  min-width: 100px
  margin: 0 0 20px 0
  text-align: center

  @media(min-width: $width-mobile)
    margin: 0 46px 0 0
    text-align: left
