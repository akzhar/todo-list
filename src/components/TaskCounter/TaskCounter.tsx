import React from 'react';

type TTaskCounterProps = {
  count: number,
  verb?: string
};

const TaskCounter: React.FC<TTaskCounterProps>= ({ count, verb = '' }) => (
  <span className="task-counter">{`${count} item${count !== 1 ? 's' : ''} ${verb}`}</span>
);

export default TaskCounter;
