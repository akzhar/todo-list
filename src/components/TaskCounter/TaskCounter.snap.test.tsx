import React from 'react';
import { test, describe, expect } from '@jest/globals';
import { render } from '@testing-library/react';

import TaskCounter from '@components/TaskCounter';

describe('TaskCounter should render correct active task count', () => {

  test('If 0 task left, it should be in the plural', () => {
    const { container } = render(<TaskCounter count={0} />);
    expect(container).toMatchSnapshot();
  });

  test('If 1 task left, it should be in the singular', () => {
    const { container } = render(<TaskCounter count={1} />);
    expect(container).toMatchSnapshot();
  });

  test('If 2 task left, it should be in the plural', () => {
    const { container } = render(<TaskCounter count={2} />);
    expect(container).toMatchSnapshot();
  });

  test('If 12 task left, it should be in the plural', () => {
    const { container } = render(<TaskCounter count={12} />);
    expect(container).toMatchSnapshot();
  });

});

