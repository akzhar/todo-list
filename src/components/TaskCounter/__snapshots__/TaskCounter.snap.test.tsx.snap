// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskCounter should render correct active task count If 0 task left, it should be in the plural 1`] = `
<div>
  <span
    class="task-counter"
  >
    0 items 
  </span>
</div>
`;

exports[`TaskCounter should render correct active task count If 1 task left, it should be in the singular 1`] = `
<div>
  <span
    class="task-counter"
  >
    1 item 
  </span>
</div>
`;

exports[`TaskCounter should render correct active task count If 2 task left, it should be in the plural 1`] = `
<div>
  <span
    class="task-counter"
  >
    2 items 
  </span>
</div>
`;

exports[`TaskCounter should render correct active task count If 12 task left, it should be in the plural 1`] = `
<div>
  <span
    class="task-counter"
  >
    12 items 
  </span>
</div>
`;
